epilepsy12/common_view_functions/aggregate_by.py

```diff
@@ -345,80 +345,173 @@ def update_kpi_aggregation_model(
     
     # update models where numbers have changed.
     for value_count in kpi_value_counts:
-        ABSTRACTION_CODE = value_count.pop(f"organisation__{abstraction_level.value}")
-        if ABSTRACTION_CODE is None:
-            # we don't have any values for this abstraction level (eg local health board only applies in Wales not England)
-            return
+        ABSTRACTION_CODE = value_count.pop(f"organisation__{abstraction_level.value}") #value_count.get(next(iter(value_count))) # 
+        if ABSTRACTION_CODE is not None:
+            # the last value in each abstraction_level is None - discard
```

Comment:
I agree with removing the early return but maybe the comment is clearer above the if?

I also don't think the last value is None, I think it's if we're grouping by an abstraction level not relevant here eg by local health board in England